### ISSUE TITLE
Jansson Module

### ISSUE THREAD

**Issue by nicoonline** [kamailio/kamailio#1978]
Hello,

is there a way to add Jansson module for Centos 6 packages  ?

**Comment by nicoonline** [comment]
jansson is available in Centos 6

Jun 10 08:41:15 proxy1 yum[21346]: Installed: jansson-devel-doc-2.11-1.el6.noarch
Jun 10 08:41:15 proxy1 yum[21346]: Installed: jansson-devel-2.11-1.el6.x86_64

[root@proxy1 ~]# locate jansson
/usr/include/jansson.h
/usr/include/jansson_config.h
/usr/lib64/libjansson.so
/usr/lib64/libjansson.so.4
/usr/lib64/libjansson.so.4.11.0

**Comment by sergey-safarov** [comment]
I will check and add packaging for CentOS 6.

**Comment by nicoonline** [comment]
Hello,
Any updates.

**Comment by miconda** [comment]
Was there a rebuild of rpms triggered due to this issue here? It was reported on mailing list that they are not available anymore:

  * https://lists.kamailio.org/pipermail/sr-users/2019-June/105906.html

Could be the opensuse infrastructure, just trying to track the reasons ...

**Comment by sergey-safarov** [comment]
Hi Daniel
Rebuild trigered by issue on opensuse infrastructure.
But last rebuild I also tried build `jansson` module on `CentOS 6` and get this error
```
[  246s] make[3]: `libkamailio_ims.so.0.1' is up to date.
[  246s] LD (gcc) [M ims_ipsec_pcscf.so]		ims_ipsec_pcscf.so
[  246s] CC (gcc) [M jansson.so]		jansson_funcs.o
[  246s] CC (gcc) [M jansson.so]		jansson_mod.o
[  246s] CC (gcc) [M jansson.so]		jansson_path.o
[  246s] CC (gcc) [M jansson.so]		jansson_utils.o
[  246s] LD (gcc) [M jansson.so]		jansson.so
[  246s] CC (gcc) [M acc_json.so]		acc_json_mod.o
[  246s] LD (gcc) [M acc_json.so]		acc_json.so
[  247s] Package libevent was not found in the pkg-config search path.
[  247s] Perhaps you should add the directory containing `libevent.pc'
[  247s] to the PKG_CONFIG_PATH environment variable
[  247s] No package 'libevent' found
[  247s] Package libevent was not found in the pkg-config search path.
[  247s] Perhaps you should add the directory containing `libevent.pc'
[  247s] to the PKG_CONFIG_PATH environment variable
[  247s] No package 'libevent' found
[  247s] Package libevent was not found in the pkg-config search path.
[  247s] Perhaps you should add the directory containing `libevent.pc'
[  247s] to the PKG_CONFIG_PATH environment variable
[  247s] No package 'libevent' found
[  247s] Package libevent was not found in the pkg-config search path.
[  247s] Perhaps you should add the directory containing `libevent.pc'
[  247s] to the PKG_CONFIG_PATH environment variable
[  247s] No package 'libevent' found
[  247s] Package libevent was not found in the pkg-config search path.
[  247s] Perhaps you should add the directory containing `libevent.pc'
[  247s] to the PKG_CONFIG_PATH environment variable
[  247s] No package 'libevent' found
[  247s] Package libevent was not found in the pkg-config search path.
[  247s] Perhaps you should add the directory containing `libevent.pc'
[  247s] to the PKG_CONFIG_PATH environment variable
[  247s] No package 'libevent' found
[  247s] CC (gcc) [M janssonrpcc.so]		janssonrpc_connect.o
[  247s] In file included from janssonrpc_connect.c:40:
[  247s] netstring.h:27:32: warning: event2/bufferevent.h: No such file or directory
[  247s] In file included from janssonrpc_io.h:31,
[  247s]                  from janssonrpc_request.h:26,
[  247s]                  from janssonrpc_connect.c:42:
[  247s] janssonrpc_server.h:29:27: warning: event2/buffer.h: No such file or directory
[  247s] janssonrpc_connect.c: In function 'bev_disconnect':
[  247s] janssonrpc_connect.c:59: warning: implicit declaration of function 'bufferevent_get_enabled'
[  247s] janssonrpc_connect.c: In function 'server_backoff_cb':
[  247s] janssonrpc_connect.c:143: warning: implicit declaration of function 'event_free'
[  247s] janssonrpc_connect.c: In function 'wait_server_backoff':
[  247s] janssonrpc_connect.c:175: warning: implicit declaration of function 'evtimer_new'
[  247s] janssonrpc_connect.c:175: warning: assignment makes pointer from integer without a cast
[  247s] janssonrpc_connect.c: In function 'bev_connect_cb':
[  247s] janssonrpc_connect.c:273: error: 'BEV_EVENT_ERROR' undeclared (first use in this function)
[  247s] janssonrpc_connect.c:273: error: (Each undeclared identifier is reported only once
[  247s] janssonrpc_connect.c:273: error: for each function it appears in.)
[  247s] janssonrpc_connect.c:273: error: 'BEV_EVENT_EOF' undeclared (first use in this function)
[  247s] janssonrpc_connect.c:276: warning: implicit declaration of function 'bufferevent_socket_get_dns_error'
[  247s] janssonrpc_connect.c:278: warning: implicit declaration of function 'evutil_gai_strerror'
[  247s] janssonrpc_connect.c:278: warning: format '%s' expects type 'char *', but argument 12 has type 'int'
[  247s] janssonrpc_connect.c:278: warning: format '%s' expects type 'char *', but argument 10 has type 'int'
[  247s] janssonrpc_connect.c:283: error: 'BEV_EVENT_CONNECTED' undeclared (first use in this function)
[  247s] janssonrpc_connect.c: In function 'bev_connect':
[  247s] janssonrpc_connect.c:362: warning: implicit declaration of function 'bufferevent_socket_new'
[  247s] janssonrpc_connect.c:365: error: 'BEV_OPT_CLOSE_ON_FREE' undeclared (first use in this function)
[  247s] janssonrpc_connect.c:379: warning: implicit declaration of function 'bufferevent_socket_connect_hostname'
[  247s] make[2]: *** [janssonrpc_connect.o] Error 1
[  247s] make[1]: *** [modules] Error 1
[  247s] make[1]: Leaving directory `/home/abuild/rpmbuild/BUILD/kamailio-5.3.0-dev5/src'
[  247s] make: *** [every-module] Error 2
[  247s] error: Bad exit status from /var/tmp/rpm-tmp.02K8aC (%build)
[  247s] 
```

I checked presence of `libevent.pc` and not found on CentOS 6. `libevent-devel` package is installed on system.
```sh
[root@628f362463dd /]# rpm -qa | grep libevent-devel
libevent-devel-1.4.13-4.el6.x86_64
[root@628f362463dd /]# find / -name libevent.pc
[root@628f362463dd /]# 
```
Also looked for `bufferevent.h`. This file also not exist on system.
```sh
[root@628f362463dd /]# find / -name bufferevent.h
[root@628f362463dd /]# 
```

Then i checked `libevent-devel` version on CentOS 7. Here is used 2.0.21.
```
[root@027c165a1484 /]# rpm -qa | grep libevent-devel
libevent-devel-2.0.21-4.el7.x86_64
```

Think we cannot build jansson module on CentOS 6 with  `libevent-devel-1.4.13-4`.
If you not want add support of  `libevent-devel-1.4.13-4` into kamailio, then please close ticket.

**Comment by nicoonline** [comment]
Hello,
Are you sure your system is UpToDate ?

[root@proxy1 ~]# find / -name libevent.pc
/usr/lib64/pkgconfig/libevent.pc

[root@proxy1 ~]# find / -name bufferevent.h
/usr/include/event2/bufferevent.h


De : sergey-safarov <notifications@github.com>
Envoyé : jeudi 13 juin 2019 10:43
À : kamailio/kamailio <kamailio@noreply.github.com>
Cc : Nicolas Breuer <Nicolas.Breuer@belcenter.biz>; Author <author@noreply.github.com>
Objet : Re: [kamailio/kamailio] Jansson Module (#1978)


Hi Daniel
Rebuild trigered by issue on opensuse infrastructure.
But last rebuild I also tried build jansson module on CentOS 6 and get this error

[  246s] make[3]: `libkamailio_ims.so.0.1' is up to date.

[  246s] LD (gcc) [M ims_ipsec_pcscf.so]             ims_ipsec_pcscf.so

[  246s] CC (gcc) [M jansson.so]              jansson_funcs.o

[  246s] CC (gcc) [M jansson.so]              jansson_mod.o

[  246s] CC (gcc) [M jansson.so]              jansson_path.o

[  246s] CC (gcc) [M jansson.so]              jansson_utils.o

[  246s] LD (gcc) [M jansson.so]              jansson.so

[  246s] CC (gcc) [M acc_json.so]             acc_json_mod.o

[  246s] LD (gcc) [M acc_json.so]             acc_json.so

[  247s] Package libevent was not found in the pkg-config search path.

[  247s] Perhaps you should add the directory containing `libevent.pc'

[  247s] to the PKG_CONFIG_PATH environment variable

[  247s] No package 'libevent' found

[  247s] Package libevent was not found in the pkg-config search path.

[  247s] Perhaps you should add the directory containing `libevent.pc'

[  247s] to the PKG_CONFIG_PATH environment variable

[  247s] No package 'libevent' found

[  247s] Package libevent was not found in the pkg-config search path.

[  247s] Perhaps you should add the directory containing `libevent.pc'

[  247s] to the PKG_CONFIG_PATH environment variable

[  247s] No package 'libevent' found

[  247s] Package libevent was not found in the pkg-config search path.

[  247s] Perhaps you should add the directory containing `libevent.pc'

[  247s] to the PKG_CONFIG_PATH environment variable

[  247s] No package 'libevent' found

[  247s] Package libevent was not found in the pkg-config search path.

[  247s] Perhaps you should add the directory containing `libevent.pc'

[  247s] to the PKG_CONFIG_PATH environment variable

[  247s] No package 'libevent' found

[  247s] Package libevent was not found in the pkg-config search path.

[  247s] Perhaps you should add the directory containing `libevent.pc'

[  247s] to the PKG_CONFIG_PATH environment variable

[  247s] No package 'libevent' found

[  247s] CC (gcc) [M janssonrpcc.so]          janssonrpc_connect.o

[  247s] In file included from janssonrpc_connect.c:40:

[  247s] netstring.h:27:32: warning: event2/bufferevent.h: No such file or directory

[  247s] In file included from janssonrpc_io.h:31,

[  247s]                  from janssonrpc_request.h:26,

[  247s]                  from janssonrpc_connect.c:42:

[  247s] janssonrpc_server.h:29:27: warning: event2/buffer.h: No such file or directory

[  247s] janssonrpc_connect.c: In function 'bev_disconnect':

[  247s] janssonrpc_connect.c:59: warning: implicit declaration of function 'bufferevent_get_enabled'

[  247s] janssonrpc_connect.c: In function 'server_backoff_cb':

[  247s] janssonrpc_connect.c:143: warning: implicit declaration of function 'event_free'

[  247s] janssonrpc_connect.c: In function 'wait_server_backoff':

[  247s] janssonrpc_connect.c:175: warning: implicit declaration of function 'evtimer_new'

[  247s] janssonrpc_connect.c:175: warning: assignment makes pointer from integer without a cast

[  247s] janssonrpc_connect.c: In function 'bev_connect_cb':

[  247s] janssonrpc_connect.c:273: error: 'BEV_EVENT_ERROR' undeclared (first use in this function)

[  247s] janssonrpc_connect.c:273: error: (Each undeclared identifier is reported only once

[  247s] janssonrpc_connect.c:273: error: for each function it appears in.)

[  247s] janssonrpc_connect.c:273: error: 'BEV_EVENT_EOF' undeclared (first use in this function)

[  247s] janssonrpc_connect.c:276: warning: implicit declaration of function 'bufferevent_socket_get_dns_error'

[  247s] janssonrpc_connect.c:278: warning: implicit declaration of function 'evutil_gai_strerror'

[  247s] janssonrpc_connect.c:278: warning: format '%s' expects type 'char *', but argument 12 has type 'int'

[  247s] janssonrpc_connect.c:278: warning: format '%s' expects type 'char *', but argument 10 has type 'int'

[  247s] janssonrpc_connect.c:283: error: 'BEV_EVENT_CONNECTED' undeclared (first use in this function)

[  247s] janssonrpc_connect.c: In function 'bev_connect':

[  247s] janssonrpc_connect.c:362: warning: implicit declaration of function 'bufferevent_socket_new'

[  247s] janssonrpc_connect.c:365: error: 'BEV_OPT_CLOSE_ON_FREE' undeclared (first use in this function)

[  247s] janssonrpc_connect.c:379: warning: implicit declaration of function 'bufferevent_socket_connect_hostname'

[  247s] make[2]: *** [janssonrpc_connect.o] Error 1

[  247s] make[1]: *** [modules] Error 1

[  247s] make[1]: Leaving directory `/home/abuild/rpmbuild/BUILD/kamailio-5.3.0-dev5/src'

[  247s] make: *** [every-module] Error 2

[  247s] error: Bad exit status from /var/tmp/rpm-tmp.02K8aC (%build)

[  247s]

I checked presence of libevent.pc and not found on CentOS 6. libevent-devel package is installed on system.

[root@628f362463dd /]# rpm -qa | grep libevent-devel

libevent-devel-1.4.13-4.el6.x86_64

[root@628f362463dd /]# find / -name libevent.pc

[root@628f362463dd /]#

Also looked for bufferevent.h. This file also not exist on system.

[root@628f362463dd /]# find / -name bufferevent.h

[root@628f362463dd /]#

Then i checked libevent-devel version on CentOS 7. Here is used 2.0.21.

[root@027c165a1484 /]# rpm -qa | grep libevent-devel

libevent-devel-2.0.21-4.el7.x86_64

Think we cannot build jansson module on CentOS 6 with libevent-devel-1.4.13-4.
If you not want add support of libevent-devel-1.4.13-4 into kamailio, then please close ticket.

**Comment by sergey-safarov** [comment]
please execute and paste here
```
rpm -qf /usr/lib64/pkgconfig/libevent.pc
```
and
```
yum whatprovides libevent-devel
```

**Comment by miconda** [comment]
So the module compiles in centos 7 but not in centos 6, due to incompatible libevent library, right?

If yes, then we can close this issue. If someone makes a PR to add support for it, then it can be reconsidered, but from my point of view, this is low priority to be considered as effort from devel team.

**Comment by sergey-safarov** [comment]
yes, that on CEntOS 6 incompatible libevent.
I will wait @nicoonline response.

**Comment by nicoonline** [comment]
libevent2

[root@proxy1 ~]# rpm -qf /usr/lib64/pkgconfig/libevent.pc
libevent2-devel-2.0.21-2.el6.x86_64

**Comment by sergey-safarov** [comment]
@nicoonline also need
```sh
yum whatprovides libevent-devel
```

**Comment by nicoonline** [comment]
libevent-devel-1.4.13-4.el6.i686 : Header files, libraries and development documentation for libevent
Repo        : base
Matched from:



libevent-devel-1.4.13-4.el6.x86_64 : Header files, libraries and development documentation for libevent
Repo        : base
Matched from:

**Comment by nicoonline** [comment]
libevent2-devel-2.0.21-2.el6.x86_64 : Header files, libraries and development documentation for libevent2
Repo        : installed
Matched from:
Other       : Provides-match: libevent2-devel

**Comment by sergey-safarov** [comment]
@nicoonline I now testing build with libevent2-devel on CentOS 6

**Comment by sergey-safarov** [comment]
Please test out kamailio repo for CentOS6.
If all ok, then i will merge SPEC file update.

**Comment by nicoonline** [comment]
yum list kamailio* | grep jan

**Comment by nicoonline** [comment]
can you put in kamailio 5.1 repos ?

**Comment by sergey-safarov** [comment]
5.1 compiled also
Please check.

**Comment by nicoonline** [comment]
installed - loaded -> OK !

**Comment by sergey-safarov** [comment]
Related commits
1. 5.1 - c03ba1f14f74b94d2ced1c79a8b44645ec636563
1. 5.2 - dcce685ab41097dc77fad50c414a66e634db72b9
1. master - a54b36e2533a71d8e38c03b497b470110a1e33fb
